### PR TITLE
First person to third person pt1

### PR DESCRIPTION
**Pull request by gurgalex**
Partial patch set for #8 
So, I divided up the commits into little pieces in case anything seems off.
Also, thanks for the existing book text!

Comments?

### CODE REVIEW

**Comment by Michael-F-Bryan** [comment]
Oh wow, thanks for taking the time to go through all of those chapters! I quickly skimmed through and from what I've seen so far I think that's definitely an improvement on what was originally written :+1: 

I'll go through and do a review to see if there's anything which can still be improved or phrased differently.

File: src/best_practices.md [hunk 1]
```diff
@@ -59,8 +59,8 @@ pub unsafe extern "C" foo_create() -> *mut Foo {
 
 Likewise, you must document all assumptions. If a function consumes the 
 resource being pointed to by a pointer then you should state that. For 
-example you might try to add some `Point` object to the `Foo` object we created
-earlier.
+example you might try to add some `Point` object to the `Foo` object just
+created earlier.
```
Comment (Michael-F-Bryan) [inline src/best_practices.md]: I feel like the "earlier" in "... just created earlier." is probably no longer necessary, seeing as "just created" means pretty much the same thing as "earlier". What are your thoughts?

File: src/callbacks/index.md [hunk 2]
```diff
@@ -198,15 +199,15 @@ impl Accumulator {
 > `*mut Accumulator`, and then turn the mutable raw pointer into a mutable
 > reference.
 
-Now our `Accumulator` struct is defined and we have a `callback()` method which
-satisfies the `StatefulCallback` function signature, we can finally get on to
-using it.
+Now our `Accumulator` struct is defined and has a `callback()` method which
+satisfies the `StatefulCallback` function signature, now, it can finally
```
Comment (Michael-F-Bryan) [inline src/callbacks/index.md]: How about rewording this to something like:

> Now that our `Accumulator` struct is defined and has a `callback()` method
> satisfying the `StatefulCallback` function signature, it is ready to be called
> by C.

I'm just trying to think of a way to avoid having the word "now" mentioned twice in the same sentence. Let me know if you can think of something better.

File: src/structs/index.md [hunk 3]
```diff
@@ -5,26 +5,26 @@ languages. As long as you tell the Rust compiler to lay out structs "how C does
 it" (with `#[repr(C)]`) and have the right type declarations, everything *Just 
 Works™*. It's all bytes at the end of the day, anyway.
 
-Remember how I mentioned earlier that there's an easier way to compile and link
-everything? Well there is, it's called [gcc-rs][gcc-rs]. All you need to do is 
+Remember that there's an easier way to compile and link everything?
```
Comment (Michael-F-Bryan) [inline src/structs/index.md]: How about, "Remember when we said there's an easier way to compile and link everything?"? The current sentence sounds a bit like it's confused whether to be a sentence or a statement.

File: src/best_practices.md (hunk 1 above)
Comment (gurgalex) [inline src/best_practices.md]: I'd agree, since the doc comment is directly below the code being mentioned.

File: src/structs/index.md (hunk 3 above)
Comment (gurgalex) [inline src/structs/index.md]: I'm trying find where the book mentioned 'gcc-rs' before.
Because if it didn't, then the paragraph could jump right in with "There's an easy way to ..."

I was trying to find a way to keep the question, but without using "we"

Comment (gurgalex) [inline src/structs/index.md]: Or, "What if there was an easy way to compile and link everything? Well, there is,..."

Comment (Michael-F-Bryan) [inline src/structs/index.md]: Yep, I like that last one :+1:

**Comment by gurgalex** [comment]
When I was editing the .md files I noticed they were ~80 characters per line.
Did you manually break each line? Or, was there a tool that determines the optimal word to cut at (for comprehension)

**Comment by Michael-F-Bryan** [comment]
I've got my editor set up to automatically wrap at 80 characters (wrapping a bit early if necessary so you don't cut words in half). So that means I can just keep typing and it'll insert a newline at the correct spot. 

I think this looks about ready to merge. Can you think of anything else which needs to be done?

**Comment by gurgalex** [comment]
I know that a few of my early commits exceed the 80 char line length before I knew about it.
Other than that, everything for this part looks fine.

**Comment by Michael-F-Bryan** [comment]
That shouldn't be an issue, it's more of a guideline than a strict rule anyway. Thanks for all the work you've put into this PR!
